### ISSUE TITLE
安装paddlepaddle-2.3.1-gpu版本，提示没有paddle.nn

### ISSUE THREAD

**Issue by chch2010523** [PaddlePaddle/Paddle#44293]
### 请提出你的问题 Please ask your question

运行gnn的例子时，提示报错Exception has occurred: ModuleNotFoundError，No module named 'paddle.nn'
我的环境是 1080TI  Ubuntu20.04
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Fri_Dec_17_18:16:03_PST_2021
Cuda compilation tools, release 11.6, V11.6.55
Build cuda_11.6.r11.6/compiler.30794723_0
NVIDIA-SMI 510.39.01    Driver Version: 510.39.01    CUDA Version: 11.6 
在conda中安装了paddlepaddle-gpu版本2.3.1
paddle-bfloat             0.1.7                    pypi_0    pypi
paddlehelix               1.0.1                    pypi_0    pypi
paddlepaddle-gpu          2.3.1.post116            pypi_0    pypi
pandas                    1.3.5                    pypi_0    pypi
pgl                       2.2.3.post0              pypi_0    pypi
我的代码是：
import paddle
import paddle.nn as nn

__all__ = ['LinearBlock', 'MLP']

class LinearBlock(nn.Layer):
    """
    Implementation of a linear block: linear-BatchNorm-ReLU

    Args:
        input_size(int): size of the input
        output_size(int): size of the output

    Examples:
        ..code-block:: python

            train_input = paddle.to_tensor([[0.0, 0.0], [1.0, 1.0]])
            mlp = paddle.nn.Sequential(
                LinearBlock(2, 5),
                LinearBlock(5, 1)
            )
            print(mlp(train_input)) # A paddle.Tensor with shape=[2, 1]

    """

    init_weight_attr = paddle.framework.ParamAttr(initializer=nn.initializer.TruncatedNormal(mean=0.0, std=0.01))
    init_bias_attr = paddle.framework.ParamAttr(initializer=nn.initializer.Constant(value=0.0))

    def __init__(self, input_size, output_size):
        super(LinearBlock, self).__init__()
        # Linear layer
        self.linear = nn.Linear(input_size,
                                output_size,
                                weight_attr=self.init_weight_attr,
                                bias_attr=self.init_bias_attr)
        # Batch-normalization layer
        self.norm = nn.BatchNorm1D(output_size, momentum=0.999, epsilon=1e-3)
        # ReLU activation
        self.relu = nn.ReLU()

    def forward(self, inputs):
        x = self.linear(inputs)
        x = self.norm(x)
        return self.relu(x)

class MLP(nn.Layer):
    """
    Implementation of MLP(Multi-Layer Perceptron) model.

    Args:
        input_size(int or tuple): size of the input, int for a vector and tuple for an image. As MLP flattens the
            input, users can also set input_size to the absolut size of the image. For example, if the input is a
            28 * 28, 3-channel image, both (28, 28, 3) and 28 * 28 * 3 is the legal input_size.
        output_size(int): size of the output.
        hidden_sizes(List, optional): hidden sizes of each hidden layer, default None. If None, it will be set
            [256, 128, 64, 64]. Please note that there is a final classifier after all hidden layers, so the last
            item of hidden_sizes is not the output size.

    Examples:
        ..code-block:: python

            import paddle
            from paddlefsl.vision.backbones import MLP
            train_input = paddle.to_tensor([[0.0, 0.0], [1.0, 1.0]])
            mlp = MLP(input_size=2, output_size=1, hidden_sizes=[5])
            print(mlp(train_input)) # A paddle.Tensor with shape=[2, 1]

    """

    init_weight_attr = paddle.framework.ParamAttr(initializer=nn.initializer.TruncatedNormal(mean=0.0, std=0.01))
    init_bias_attr = paddle.framework.ParamAttr(initializer=nn.initializer.Constant(value=0.0))

    def __init__(self, input_size, output_size, hidden_sizes=None):
        super(MLP, self).__init__()
        # Flatten layer
        self.flatten = nn.Flatten()
        # Hidden layers
        if type(input_size) is not int:
            hidden_in = 1
            for size in input_size:
                hidden_in *= size
        else:
            hidden_in = input_size
        hidden_sizes = [256, 128, 64, 64] if hidden_sizes is None else hidden_sizes
        hidden_out = hidden_sizes[0]
        self.hidden = nn.Sequential(
            ('hidden0', LinearBlock(hidden_in, hidden_out))
        )
        for i in range(1, len(hidden_sizes)):
            hidden_in, hidden_out = hidden_out, hidden_sizes[i]
            self.hidden.add_sublayer(name='hidden' + str(i),
                                     sublayer=LinearBlock(hidden_in, hidden_out))
        # Output layer
        self.output = paddle.nn.Linear(hidden_sizes[-1],
                                       output_size,
                                       weight_attr=self.init_weight_attr,
                                       bias_attr=self.init_bias_attr)

    def forward(self, inputs):
        y = self.flatten(inputs)
        y = self.hidden(y)
        return self.output(y)

**Comment by kuizhiqing** [comment]
请检查使用的 python 版本对应的按照目录下 paddle 是否安装正确，另外工作目录下是否包含了 paddle 目录导致 import 首先识别当前目录。

**Comment by chch2010523** [comment]
检查后，确认当前目录下没有paddle目录。
虚拟环境python版本为3.7，使用conda list检查得到如下结果

```
# packages in environment at /home/cc/miniconda3/envs/paddlefsl37:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
_openmp_mutex             4.5                       2_gnu    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
astor                     0.8.1                    pypi_0    pypi
ca-certificates           2022.6.15            ha878542_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
certifi                   2022.6.15                pypi_0    pypi
charset-normalizer        2.0.12                   pypi_0    pypi
cython                    0.29.30                  pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
idna                      3.3                      pypi_0    pypi
joblib                    1.1.0                    pypi_0    pypi
ld_impl_linux-64          2.36.1               hea4e1c9_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libffi                    3.4.2                h7f98852_5    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgcc-ng                 12.1.0              h8d9b700_16    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgomp                   12.1.0              h8d9b700_16    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libnsl                    2.0.0                h7f98852_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libstdcxx-ng              12.1.0              ha89aaad_16    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libzlib                   1.2.12               h166bdaf_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
ncurses                   6.3                  h27087fc_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
networkx                  2.6.3                    pypi_0    pypi
numpy                     1.19.5                   pypi_0    pypi
onnx                      1.12.0                   pypi_0    pypi
opencv-python             4.6.0.66                 pypi_0    pypi
openssl                   3.0.4                h166bdaf_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
opt-einsum                3.3.0                    pypi_0    pypi
paddle-bfloat             0.1.7                    pypi_0    pypi
paddlehelix               1.0.1                    pypi_0    pypi
paddlepaddle-gpu          2.3.1.post116            pypi_0    pypi
pandas                    1.3.5                    pypi_0    pypi
pgl                       2.2.3.post0              pypi_0    pypi
pillow                    9.1.1                    pypi_0    pypi
pip                       22.1.2             pyhd8ed1ab_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
protobuf                  3.20.0                   pypi_0    pypi
python                    3.7.12          hf930737_100_cpython    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.7                     2_cp37m    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
pytz                      2022.1                   pypi_0    pypi
rdkit                     2022.3.3                 pypi_0    pypi
readline                  8.1.2                h0f457ee_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
requests                  2.28.0                   pypi_0    pypi
scikit-learn              1.0.2                    pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
setuptools                62.6.0           py37h89c1867_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
six                       1.16.0                   pypi_0    pypi
sklearn                   0.0                      pypi_0    pypi
sqlite                    3.39.0               h4ff8645_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
thop                      0.1.0-2207010342          pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tk                        8.6.12               h27826a3_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
torch                     1.12.0+cu116             pypi_0    pypi
torchaudio                0.12.0+cu116             pypi_0    pypi
torchvision               0.13.0+cu116             pypi_0    pypi
tqdm                      4.64.0                   pypi_0    pypi
typing-extensions         4.2.0                    pypi_0    pypi
ujson                     5.3.0                    pypi_0    pypi
urllib3                   1.26.9                   pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
xz                        5.2.5                h516909a_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
zlib                      1.2.12               h166bdaf_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
```
安装命令使用的是官网的
`python -m pip install paddlepaddle-gpu==2.3.1.post116 -f https://www.paddlepaddle.org.cn/whl/linux/mkl/avx/stable.html`

**Comment by kuizhiqing** [comment]
使用环境和安装环境需要对应，你目前使用 conda 环境安装，需要通过 conda 环境来使用。具体使用方法请参考 conda 文档。另外建议可以尝试使用 docker，简化环境配置工作。
